test/integration/framework/cluster.go

```
@@ -173,6 +200,141 @@ func (c *Cluster) GetLogs(ctx context.Context, resourceRef string, tailLines int
+
+// PortForward starts a port-forward to the given resource (e.g. "svc/my-service")
+// and remote port. It returns the local port assigned by the OS and a cancel
+// function that stops the port-forward. The caller must call cancel when done.
+func (c *Cluster) PortForward(resourceRef string, remotePort int) (localPort int, cancel func(), err error) {
+	const startupTimeout = 30 * time.Second
+
+	args := c.addNamespace([]string{"port-forward", resourceRef, fmt.Sprintf("0:%d", remotePort)})
+	startupCtx, startupCancel := context.WithTimeout(context.Background(), startupTimeout)
+	defer startupCancel()
+
+	cmd := exec.Command("oc", args...)
```

golangci-lint (2.11.4)

[error] os/exec.Command must not be called. use os/exec.CommandContext

(noctx)